multi-layer-perceptron/KNet_test.jl

```diff
@@ -0,0 +1,143 @@
+using MLDatasets: MNIST
+using Knet, IterTools, MLDatasets
+using Dictionaries
+using TimerOutputs
+using TimerOutputs
+using JSON
+using Printf
+using Knet:minibatch
+using Knet:minimize
+using Knet
+using Knet: Param
+using Knet: Knet, dir, accuracy, progress, sgd, gc, Data, nll, relu
+using Flatten
+using Flux.Data;
+using Flux, Statistics
+
+# This loads the MNIST handwritten digit recognition dataset. This code is based off the Knet Tutorial Notebook. 
+xtrn,ytrn = MNIST.traindata(Float32)
+xtst,ytst = MNIST.testdata(Float32)
+println.(summary.((xtrn,ytrn,xtst,ytst)));
+
+xtrn = reshape(xtrn, 784, 60000 ) 
+xtst = reshape(xtst, 784, 10000 )
+println(summary.((xtrn, xtst))) # can see the data that is flattened 
+
+#Preprocessing targets: one hot vectors
+# ytrn = onehotbatch(ytrn, 0:9)
+# ytst = onehotbatch(ytst, 0:9)
+
+train_loader = DataLoader((xtrn, ytrn), batchsize=128);
+test_loader = DataLoader((xtst, ytst), batchsize = 128)
+
+length(test_loader)
+
+
+
+(x,y) = first(train_loader) #gives the first minibatch from training dataset
+println.(summary.((x,y)));
+
+
+
+
+
+struct Dense1; w; b; f; end
+Dense1(i,o; f=relu) = Dense1(param(o,i), param0(o), f)
+(d::Dense1)(x) = d.f.(d.w * mat(x) .+ d.b)
+
+# Define a chain of layers and a loss function:
+struct Chain; layers; end
+(c::Chain)(x) = (for l in c.layers; x = l(x); end; x)
+(c::Chain)(x,y) = nll(c(x),y)
+
+model = Chain((Dense1(784, 100), Dense1(100, 10), identity))
+
+model(x) #checking if training is working
```

Comment:
If you call model(x) here in the script, it won't be included in the jit time in the training loop. I would comment out this line in the script, but it's fine for the notebook.